packages/core/src/lib/waku_store/index.ts

```diff
@@ -382,6 +389,33 @@ export function isDefined<T>(msg: T | undefined): msg is T {
   return !!msg;
 }
 
+export async function createCursor(
+  message: DecodedMessage,
+  pubsubTopic: string = DefaultPubSubTopic
+): Promise<Index> {
+  if (
+    !message ||
+    !message.timestamp ||
+    !message.payload ||
+    !message.contentTopic
+  ) {
+    throw new Error("Message is missing required fields");
+  }
+
+  const contentTopicBytes = utf8ToBytes(message.contentTopic);
+
+  const digest = sha256(concat([contentTopicBytes, message.payload]));
+
+  const messageTime = BigInt(message.timestamp.getTime()) * BigInt(1000000);
+
+  return {
+    digest,
+    pubsubTopic,
+    senderTime: messageTime,
+    receivedTime: messageTime,
```

Comment:
@danisharora099 @felicio is right here. I don't thinks the `proto.Index` interface should be directly used in the API because the API consumer will never know the `receivedTime` value.

I suggest we create an `Index` (or even better, `Cursor`) interface that contains all field above apart from `receivedTime`.